internal/netxlite/tlshandshaker.go

```diff
@@ -26,6 +28,10 @@ type TLSHandshakerConfigurable struct {
 	// Timeout is the OPTIONAL timeout imposed on the TLS handshake. If zero
 	// or negative, we will use default timeout of 10 seconds.
 	Timeout time.Duration
+
+	// UTLSHandshaker is the OPTIONAL Handshaker that uses utls for TLS parroting
+	// if this Handshaker is not set, we'll use net
+	UTLSHandshaker *UTLSHandshaker
```

Comment:
Rather than adding an `UTLSHandshaker` type, could you instead provide an implementation of `NewConn` that returns a connection that implements `TLSConn` and uses `*utls.UConn` as the underlying conn?